src/common/config.cc

```diff
@@ -1236,21 +1236,45 @@ bool md_config_t::expand_meta(std::string &origval,
 }
 
 void md_config_t::diff(
+  const md_config_t *other,
+  map<string, pair<string, string> > *diff,
+  set<string> *unknown) 
+{
+  diff_helper(other, diff, unknown, "");
+}
+void md_config_t::diff(
+  const md_config_t *other,
+  map<string, pair<string, string> > *diff,
+  set<string> *unknown, const string& ceph_setting = string("")) 
+{
+  diff_helper(other, diff, unknown, ceph_setting);
+}
+
+void md_config_t::diff_helper(
     const md_config_t *other,
     map<string,pair<string,string> > *diff,
-    set<string> *unknown)
+    set<string> *unknown, const string& ceph_setting=string(""))
 {
   Mutex::Locker l(lock);
 
+  bool ceph_param_search = (!ceph_setting.empty()) ? true : false;
   char local_buf[4096];
   char other_buf[4096];
   for (auto& opt: *config_options) {
+    if (ceph_param_search) {
+      if (ceph_setting != opt.name) {
+        continue;
+      }
+    }
     memset(local_buf, 0, sizeof(local_buf));
     memset(other_buf, 0, sizeof(other_buf));
 
     char *other_val = other_buf;
     int err = other->get_val(opt.name, &other_val, sizeof(other_buf));
     if (err < 0) {
+      if (ceph_param_search) {
```

Comment:
If we didn't find the param, why spend time going through the rest of the code, as opposed to spend time looking at the next one?